src/swaps/sovryn/SovrynSwapProvider.js

```diff
@@ -0,0 +1,372 @@
+import BN from 'bignumber.js'
+import { v4 as uuidv4 } from 'uuid'
+import * as ethers from 'ethers'
+
+import { chains, currencyToUnit, unitToCurrency } from '@liquality/cryptoassets'
+import cryptoassets from '@/utils/cryptoassets'
+import { isERC20 } from '../../utils/asset'
+import { prettyBalance } from '../../utils/coinFormatter'
+import { ChainNetworks } from '../../store/utils'
+import { withInterval, withLock } from '../../store/actions/performNextAction/utils'
+import { SwapProvider } from '../SwapProvider'
+import ERC20 from '@uniswap/v2-core/build/ERC20.json'
+
+import SovrynSwapNetworkABI from './abiSovrynSwapNetwork'
+import RBTCWrapperProxyABI from './abiRBTCWrapperProxy'
+
+// use WRBTC address for RBTC native token
+const nativeAssetAddress = {
+  mainnet: '0x542fDA317318eBF1d3DEAf76E0b632741A7e677d',
+  testnet: '0x69FE5cEC81D5eF92600c1A0dB1F11986AB3758Ab'
+}
+
+class SovrynSwapProvider extends SwapProvider {
+  constructor (config) {
+    super(config)
+    this._apiCache = {} // chainId to RPC provider
+  }
+
+  async getSupportedPairs () {
+    return []
+  }
+
+  // returns rates between tokens
+  async getQuote ({ network, from, to, amount }) {
+    const fromInfo = cryptoassets[from]
+    const toInfo = cryptoassets[to]
+
+    // only RSK network swaps
+    if (fromInfo.chain !== 'rsk' || toInfo.chain !== 'rsk' || amount < 0) return null
+
+    const fromAddress = (fromInfo.contractAddress || nativeAssetAddress[network]).toLowerCase()
+    const toAddress = (toInfo.contractAddress || nativeAssetAddress[network]).toLowerCase()
+    const fromAmountInUnit = currencyToUnit(fromInfo, BN(amount))
+
+    const ssnContract = new ethers.Contract(this.config.routerAddress.toLowerCase(), SovrynSwapNetworkABI, this._getApi(network, from))
+
+    // generate path
+    const path = await ssnContract.conversionPath(fromAddress, toAddress)
+    // calculate rates
+    const rate = await ssnContract.rateByPath(path, fromAmountInUnit.toString(10))
```

Comment:
When base is added to .toString it will return normal notation. Thats why I'm just adding the default 10 base.

     * Regardless of the value of `EXPONENTIAL_AT`, the `toFixed` method will always return a value in
     * normal notation and the `toExponential` method will always return a value in exponential form.
     * Calling `toString` with a base argument, e.g. `toString(10)`, will also always return normal
     * notation.

Reply:
toString(10) is replaced with toFixed()